### ISSUE TITLE
XML special characters escaping

### ISSUE THREAD

**Issue by NozomiClambie** [HDoujinDownloader/HDoujinDownloader#54]
As per https://doujindownloader.com/forum/viewtopic.php?f=9&t=1204 which has amounted quite a few views but no responses and the email I sent (I don't know what your turnaround time is but I figured documenting it on github wouldn't hurt)

XML needs the following characters to be escaped
| Literal Character | Escaped Form |
| ----------- | ----------- |
| `"` | `&quot;` |
| `'`   | `&apos;` |
| `<`   | `&lt;` |
| `>`   | `&gt;` |
| `&`   | `&amp;` |

Trying to make use of any XML files that contains one of these special characters without it being properly escaped will cause problems, typically in the form of throwing an error about the entire XML file being malformed or invalid
https://www.w3.org/TR/2008/REC-xml-20081126/#syntax

**Comment by HDoujinDownloader** [comment]
Thanks for bringing this up again! The forum post in question must've slipped off my radar.

Honestly, when XML support was added many years ago, I actually implemented the serialization process manually resulting in a _very_ naïve solution. I'll write a proper solution later, but for now I've made changes to escape each field before it's written to the XML document. These changes will be present in the next update!
